### PR TITLE
feat(apiconnect): initial implemention of ApiConnectComponent

### PR DESCRIPTION
**Pull request by raymondfeng**
This PR adds an extension for IBM API Connect to enhance the OpenAPI spec with `x-ibm-configuration` extension.

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [x] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: extensions/apiconnect/src/keys.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/openapi-v3
```
Comment (derdeka) [inline extensions/apiconnect/src/keys.ts]:
```suggestion
// Node module: @loopback/apiconnect
```

Comment (raymondfeng) [reply extensions/apiconnect/src/keys.ts]: Good catch! Fixed.

File: extensions/apiconnect/README.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+- [Guidelines](https://github.com/strongloop/loopback-next/blob/master/docs/CONTRIBUTING.md)
+- [Join the team](https://github.com/strongloop/loopback-next/issues/110)
+
+## Tests
+
+Run `npm test` from the root folder.
+
+The acceptance test against fluentd is available as a separate package at
+`acceptance/extension-logging-fluentd`.
```
Comment (achrinza) [inline extensions/apiconnect/README.md]:
```suggestion
```

Comment (raymondfeng) [reply extensions/apiconnect/README.md]: Good catch. It's a copy/paste error. Fixed.

File: extensions/apiconnect/README.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
+# @loopback/apiconnect
+
+This module extends LoopBack with ability to integrate with
```
Comment (dhmlau) [inline extensions/apiconnect/README.md]:
```suggestion
This module extends LoopBack with the ability to integrate with
```

File: extensions/apiconnect/src/__tests__/acceptance/apiconnect.acceptance.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Application} from '@loopback/core';
+import {RestApplication, RestServer} from '@loopback/rest';
+import {expect} from '@loopback/testlab';
+import {ApiConnectBindings, ApiConnectComponent} from '../..';
+import {ApiConnectSpecOptions} from '../../apiconnect.spec-enhancer';
+
+describe('LoopBack Extension for IBM API Connect', () => {
```
Comment (dhmlau) [inline extensions/apiconnect/src/__tests__/acceptance/apiconnect.acceptance.ts]: nitpick: making it more descriptive, I'm thinking whether it's better if the description is:
`Apply IBM API Connect Extension to LoopBack application`

File: extensions/apiconnect/README.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+```ts
+this.component(ApiConnectComponent);
+```
+
+The component requires a configuration for API Connect extension for OpenAPI
+spec:
+
+```ts
+const apiConnectOptions: ApiConnectSpecOptions = {
+  targetUrl: 'http://localhost:3000/test-service',
```
Comment (dhmlau) [inline extensions/apiconnect/README.md]: Question: if I want to use the same url for the APIC-flavored OpenAPI spec, can I not specify the `targetUrl`?

Comment (raymondfeng) [reply extensions/apiconnect/README.md]: The `targetUrl` is required as it tells API Connect gateway where the REST APIs are hosted. We cannot infer it.

**Pull request merged**